aviary/subsystems/propulsion/prop_performance.py

```diff
@@ -0,0 +1,249 @@
+                 [0.964, 0.954, 0.943, 0.912, 0.876, 0.834, 0.786],
+                 [0.955, 0.943, 0.928, 0.892, 0.848, 0.801, 0.751],
+                 [0.948, 0.935, 0.917, 0.872, 0.820, 0.763, 0.706],
+                 [0.940, 0.924, 0.902, 0.848, 0.790, 0.726, 0.662]]
+            ),
+        )
+
+        self.add_subsystem(
+            name='FT',
+            subsys=om.ExecComp(
+                'install_loss_factor = 1 - blockage_factor',
+                blockage_factor={'units': 'unitless', 'val': np.zeros(nn)},
+                install_loss_factor={'units': 'unitless', 'val': np.zeros(nn)},
+                has_diag_partials=True,
+            ),
+            promotes_inputs=["blockage_factor"],
+            promotes_outputs=["install_loss_factor"],
+        )
+
+
+class PropPerf(om.Group):
+    """
+    Computation of propeller thrust coefficient based on Hamilton Standard.
+    The installation loss factor is either a user input or computed internally.
+    """
+
+    def initialize(self):
+        self.options.declare(
+            'num_nodes', types=int, default=1,
+            desc='Number of nodes to be evaluated in the RHS')
+        self.options.declare(Aircraft.Engine.NUM_BLADES, default=2,
+                             desc='number of blades per propeller')
+        self.options.declare(
+            Aircraft.Design.COMPUTE_INSTALLATION_LOSS, types=bool,
+            desc='Flag to compute installation factor')
+        self.options.declare(
+            'compute_mach_internally', types=bool, default=False,
+        )
+        self.options.declare(
+            'aviary_options', types=AviaryValues,
+            desc='collection of Aircraft/Mission specific options')
+
+    def setup(self):
+        options = self.options
+        nn = options['num_nodes']
+        aviary_options = options['aviary_options']
+        compute_installation_loss = aviary_options.get_val(
+            Aircraft.Design.COMPUTE_INSTALLATION_LOSS)
+        num_blades = aviary_options.get_val(Aircraft.Engine.NUM_BLADES)
+
+        if compute_installation_loss:
+            self.add_subsystem(
+                name='loss',
+                subsys=InstallLoss(num_nodes=nn),
+                promotes_inputs=[
+                    Aircraft.Nacelle.AVG_DIAMETER,
+                    Aircraft.Engine.PROPELLER_DIAMETER,
+                    Dynamic.Mission.VELOCITY,
+                    Dynamic.Mission.PROPELLER_TIP_SPEED,
+                ],
+                promotes_outputs=[
+                    ("install_loss_factor", Dynamic.Mission.INSTALLATION_LOSS_FACTOR)],
+            )
+        else:
+            comp = om.IndepVarComp()
+            comp.add_output('install_loss_factor',
+                            val=np.ones(nn), units="unitless")
+            self.add_subsystem('input_install_loss', comp,
+                               promotes=[('install_loss_factor', Dynamic.Mission.INSTALLATION_LOSS_FACTOR)])
```

Comment:
could this be replaced with a set_input_defaults on `Dynamic.Mission.INSTALLATION_LOSS_FACTOR`?

Reply:
Good call, yep! Done